internal/clients/diffreviewer/circleci/circleci_service.go

```diff
@@ -58,12 +80,56 @@ func (s *Service) LoadConfig(nightfallConfigFileName string) (*nightfallconfig.C
 		return nil, errors.New("missing env var for commit sha")
 	}
 	beforeCommitSha, _ := os.LookupEnv(CircleBeforeCommitEnvVar)
+
+	var baseBranch string
+	inputBaseBranch, ok := os.LookupEnv(GithubMasterBranchEnvVar)
+	if ok && inputBaseBranch != "" {
+		baseBranch = inputBaseBranch
+	} else {
+		workflowBranch, ok := os.LookupEnv(CircleBranchEnvVar)
+		if !ok {
+			s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", CircleBranchEnvVar))
+			return nil, errors.New("missing env var for branch name")
+		}
+		// if master branch triggered the workflow, do not set the baseBranch
+		if workflowBranch != DefaultMasterBranchName {
+			baseBranch = DefaultMasterBranchName
+		}
+	}
 	s.GitDiff = &gitdiff.GitDiff{
 		WorkDir:    workspacePath,
-		BaseBranch: "master",
+		BaseBranch: baseBranch,
 		BaseSHA:    beforeCommitSha,
 		Head:       commitSha,
 	}
+
+	owner, ok := os.LookupEnv(CircleOwnerNameEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", CircleOwnerNameEnvVar))
+		return nil, errors.New("missing env var for repo owner")
+	}
+	repo, ok := os.LookupEnv(CircleRepoNameEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", CircleRepoNameEnvVar))
+		return nil, errors.New("missing env var for repository name")
+	}
+	prNumberUrl, ok := os.LookupEnv(CirclePullRequestUrlEnvVar)
+	var prNumber *int
+	if ok && prNumberUrl != "" {
```

Comment:
extract pr number from url and convert to int if `CirclePullRequestUrlEnvVar` exists, otherwise `prNumber = nil`